### PR TITLE
Fussing to get Shane Kast Blue to work 

### PR DESCRIPTION
**Pull request by profxj**
WIP but this let's @jhennawi see what I touched.

### CODE REVIEW

File: pypeit/core/arc.py [hunk 1]
```diff
@@ -1107,7 +1107,11 @@ def fit_arcspec(xarray, yarray, pixt, fitp):
 #            continue  # Probably won't be a good solution
         # Fit the gaussian
         try:
-            pypeitFit = fitting.func_fit(xarray[pmin:pmax], yarray[pmin:pmax], "gaussian", 3)#, return_errors=True)
+            # TODO -- REPLACE THIS WITH A CUSTOM GAUSSIAN FITTER
+            pypeitFit = fitting.PypeItFit(xval=xarray[pmin:pmax], yval=yarray[pmin:pmax],
```
Comment (jhennawi) [inline pypeit/core/arc.py:1111]: It is ugly to have to cast the order to an array. Can we not just check internally to PypeItFit. The easiest way would be to cast inside of PypeItFit in which case order could be an integer, a 1d array, or a 2d array or tuple for 2d fits. I realize this breaks the automated code you have that sets the attributes of the datamodel, but is it really required that all arguments to PypeItFit are part of the datamodel?

File: pypeit/core/fitting.py [hunk 2]
```diff
@@ -439,15 +442,16 @@ def robust_fit(xarray, yarray, order, x2=None, function='polynomial',
...
+    #mskcnt = np.sum(this_gpm)
     #pypeitFit = None
     while (not qdone) and (iIter < maxiter):
         if np.sum(this_gpm) <= np.sum(order) + 1:
             msgs.warn("More parameters than data points - fit might be undesirable")
         if not np.any(this_gpm):
             msgs.warn("All points were masked. Returning current fit and masking all points. Fit is likely undesirable")
 
-        pypeitFit = PypeItFit(xval=xarray, yval=yarray, func=function, order=order, x2=x2, weights=weights, gpm=this_gpm.astype(int),
+        pypeitFit = PypeItFit(xval=xarray, yval=yarray, func=function, order=np.atleast_1d(order), x2=x2, weights=weights,
```
Comment (jhennawi) [inline pypeit/core/fitting.py:453]: I am not liking this casting very much. Better to write code with flexible arguments in my opinion.

File: pypeit/core/fitting.py [hunk 3]
```diff
@@ -439,15 +442,16 @@ def robust_fit(xarray, yarray, order, x2=None, function='polynomial',
...
+                              gpm=this_gpm.astype(int),
```
Comment (jhennawi) [inline pypeit/core/fitting.py:454]: Same issue here, I'd prefer you just cast internally.

File: pypeit/core/fitting.py [hunk 4]
```diff
@@ -464,7 +468,7 @@ def robust_fit(xarray, yarray, order, x2=None, function='polynomial',
         msgs.warn('Maximum number of iterations maxiter={:}'.format(maxiter) + ' reached in robust_polyfit_djs')
 
     # Do the final fit
-    pypeitFit = PypeItFit(xval=xarray, yval=yarray, func=function, order=order, x2=x2, weights=weights, gpm=this_gpm.astype(int),
+    pypeitFit = PypeItFit(xval=xarray, yval=yarray, func=function, order=np.atleast_1d(order), x2=x2, weights=weights, gpm=this_gpm.astype(int),
```
Comment (jhennawi) [inline pypeit/core/fitting.py:471]: casting me no like.

File: pypeit/wavecalib.py [hunk 5]
```diff
@@ -192,8 +197,9 @@ def build_waveimg(self, spectrograph, tilts, slits, spat_flexure=None):
         # If this is echelle print out a status message and do some error checking
         if self.par['echelle']:
             msgs.info('Evaluating 2-d wavelength solution for echelle....')
-            if len(wv_calib['fit2d']['orders']) != np.sum(ok_slits):
-                msgs.error('wv_calib and ok_slits do not line up. Something is very wrong!')
+            # TODO UPDATE THIS!!
```
Comment (jhennawi) [inline pypeit/wavecalib.py:200]: Why did this get commented out?

**Comment by profxj** [comment]
merging.

**Pull request merged**